File: week2/readability/readability.c
#include <stdio.h>
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <string.h>

// PROTOTYPE functions count_letters, count_words and count_sentences
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // get input from the user
    string text = get_string("Text: ");

    // get number of letters, words and sentence of a text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float average_words = (float) words / 100;

    float L = (float) letters / average_words;

    float S = (float) sentences / average_words;

    // Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;

    // Convert index for int
    int indexColeman = (int) round(index);

    if (indexColeman >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (indexColeman < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", indexColeman);
    }

}


int count_letters(string text)
{
    // variable for count num of letters
    int num_letters = 0;

    // loop inside each character of text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // if character is alphabetical, sum 1 to variable num_letters
        if (isalpha(text[i]))
        {
            num_letters++;
        }
    }

    return num_letters;
}

int count_words(string text)
{
    // variable for count num of words
    // variable starts in 1 cause need count the last word in a text (wich not contains space)
    int num_words = 1;

    // loop inside each character of text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // if character is space, sum 1 to variable num_words
        if (isspace(text[i]))
        {
            num_words++;
        }
    }

    return num_words;
}

int count_sentences(string text)
{
    // variable for count num of sentences
    int num_sentences = 0;

    // loop inside each character of text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // if character is space, sum 1 to variable num_words
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            num_sentences++;
        }
    }

    return num_sentences;
}

File: week06/pset6/readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    average_words = words / 100

    L = letters / average_words

    S = sentences / average_words

    indexColeman = int(round(0.0588 * L - 0.296 * S - 15.8))

    if (indexColeman >= 16):
        print("Grade 16+")
    elif (indexColeman < 1):
        print("Before Grade 1")
    else:
        print(f"Grade {indexColeman}")


def count_letters(text):
    num_letters = 0
    n = len(text)

    for i in range(n):
        if (text[i].isalpha()):
            num_letters += 1

    return num_letters


def count_words(text):
    num_words = 1
    n = len(text)

    for i in range(n):
        if(text[i].isspace()):
            num_words += 1

    return num_words


def count_sentences(text):
    num_sentences = 0
    n = len(text)

    dot = ord('.')
    exclamation = ord('!')
    interrogation = ord('?')

    for i in range(n):
        character = ord(text[i])
        if (character == dot or character == exclamation or character == interrogation):
            num_sentences += 1

    return num_sentences


if __name__ == "__main__":
    main()
